quill-core/src/main/scala/io/getquill/ast/StatefulTransformer.scala

```diff
@@ -37,6 +39,9 @@ trait StatefulTransformer[T] {
         (If(at, bt, ct), ctt)
 
       case l: Dynamic => (l, this)
+
+      case e @ (_: Block | _: Val) =>
```

Comment:
The transformers should work with all AST types, why not support `Block` and `Val`?